### PR TITLE
Fix media streaming bug

### PR DESCRIPTION
**Pull request by srinathnarayanan**
- Removed media streaming related checks in synchronized_request that was causing issues with DBAs with "media" in the name
- Refactored pipeline_client.run into a _PipelineRun method for testing purposes
- Removed all attachment and media APIs which are unused in the cosmos_client_connection
- Removed MediaReadmode and MediaRequestTimeout

### CODE REVIEW

**Comment by adxsdk6** [comment]
Can one of the admins verify this patch?

File: sdk/cosmos/azure-cosmos/azure/cosmos/_synchronized_request.py [hunk 1]
```diff
@@ -90,10 +90,7 @@ def _Request(global_endpoint_manager, request_params, connection_policy, pipelin
     """
     # pylint: disable=protected-access
 
-    is_media = request.url.find("media") > -1
-    is_media_stream = is_media and connection_policy.MediaReadMode == documents.MediaReadMode.Streamed
-
-    connection_timeout = connection_policy.MediaRequestTimeout if is_media else connection_policy.RequestTimeout
```
Comment (annatisch) [inline sdk/cosmos/azure-cosmos/azure/cosmos/_synchronized_request.py]: Could we remove `MediaRequestTimeout` from the connection policy object and documentation?
Some other places to remove:
https://github.com/Azure/azure-sdk-for-python/blob/master/sdk/cosmos/azure-cosmos/azure/cosmos/cosmos_client.py#L80
https://github.com/Azure/azure-sdk-for-python/blob/master/sdk/cosmos/azure-cosmos/azure/cosmos/cosmos_client.py#L82
https://github.com/Azure/azure-sdk-for-python/blob/master/sdk/cosmos/azure-cosmos/azure/cosmos/cosmos_client.py#L133
https://github.com/Azure/azure-sdk-for-python/blob/master/sdk/cosmos/azure-cosmos/azure/cosmos/cosmos_client.py#L136
https://github.com/Azure/azure-sdk-for-python/blob/master/sdk/cosmos/azure-cosmos/azure/cosmos/documents.py#L341
https://github.com/Azure/azure-sdk-for-python/blob/master/sdk/cosmos/azure-cosmos/azure/cosmos/documents.py#L347

File: sdk/cosmos/azure-cosmos/azure/cosmos/_synchronized_request.py [hunk 2]
```diff
@@ -90,10 +90,7 @@ def _Request(global_endpoint_manager, request_params, connection_policy, pipelin
     """
     # pylint: disable=protected-access
 
-    is_media = request.url.find("media") > -1
-    is_media_stream = is_media and connection_policy.MediaReadMode == documents.MediaReadMode.Streamed
```
Comment (annatisch) [inline sdk/cosmos/azure-cosmos/azure/cosmos/_synchronized_request.py]: We should remove the MediaReadMode enum from documents.py

Comment (srinathnarayanan) [reply sdk/cosmos/azure-cosmos/azure/cosmos/_synchronized_request.py]: we should also move all media related api s on cosmos_client_connection, will do that in the next iteration

File: sdk/cosmos/azure-cosmos/HISTORY.md [hunk 3]
```diff
@@ -1,5 +1,9 @@
 # Release History
 
+## 4.0.0b6
+
+- Fixed bug in synchronized_request for media APIs.
```
Comment (annatisch) [inline sdk/cosmos/azure-cosmos/HISTORY.md]: We also need to add a list of the breaking changes, specifically the removal of the media configuration options and the MediaReadMode.

Comment (annatisch) [reply sdk/cosmos/azure-cosmos/HISTORY.md]: @srinathnarayanan - after this last update to the release notes we should be able to merge :)

**Pull request merged**